lib/sparse/src/index/inverted_index.rs

```diff
@@ -0,0 +1,49 @@
+use std::collections::HashMap;
+
+use crate::common::types::RecordId;
+use crate::index::posting_list::PostingList;
+
+pub struct InvertedIndex {
+    pub postings: Vec<PostingList>,
+}
+
+impl InvertedIndex {
+    pub fn get(&self, id: &RecordId) -> Option<&PostingList> {
+        self.postings.get((*id) as usize)
+    }
+}
+
+pub struct InvertedIndexBuilder {
+    postings: HashMap<u32, PostingList>,
+}
+
+impl InvertedIndexBuilder {
+    pub fn new() -> InvertedIndexBuilder {
+        InvertedIndexBuilder {
+            postings: HashMap::new(),
+        }
+    }
+
+    pub fn add(&mut self, id: u32, posting: PostingList) -> &mut Self {
+        self.postings.insert(id, posting);
+        self
+    }
+
+    pub fn build(&mut self) -> InvertedIndex {
+        // Get sorted keys
+        let mut keys: Vec<u32> = self.postings.keys().copied().collect();
+        keys.sort();
```

Comment:
[`sort_unstable`](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable)?

```suggestion
        keys.sort_unstable();
```